openshift/templates/app/pims-db/mssql-deploy.yaml

```diff
@@ -0,0 +1,133 @@
+kind: Template
+apiVersion: v1
+metadata:
+  name: ${COMP_NAME}-deployment-template
+  annotations:
+    openshift.io/display-name: Microsoft SQL Server
+    description: Relational database management system developed by Microsoft.
+objects:
+  - kind: Secret
+    apiVersion: v1
+    metadata:
+      name: ${COMP_NAME}-${SUFFIX}-secret
+    stringData:
+      MSSQL_SA_PASSWORD: ${MSSQL_SA_PASSWORD}
+  - kind: Service
+    apiVersion: v1
+    metadata:
+      name: ${COMP_NAME}-${SUFFIX}
+      annotations:
+        description: Exposes the database server
+    spec:
+      ports:
+        - name: 1433-tcp
+          protocol: TCP
+          port: 1433
+          targetPort: 1433
+      selector:
+        name: ${COMP_NAME}-${SUFFIX}
+      type: ClusterIP
+      sessionAffinity: None
+  - kind: DeploymentConfig
+    apiVersion: v1
+    metadata:
+      name: ${COMP_NAME}-${SUFFIX}
+      annotations:
+        description: Defines how to deploy the database server
+    spec:
+      strategy:
+        type: Rolling
+      triggers:
+        - type: ConfigChange
+        - type: ImageChange
+          imageChangeParams:
+            automatic: true
+            containerNames:
+              - ${COMP_NAME}-${SUFFIX}
+            from:
+              kind: ImageStreamTag
+              namespace: ${TOOLS_NAMESPACE}
+              name: ${SQL_SERVER_RUNTIME_IMAGE}
+      replicas: 1
+      selector:
+        name: ${COMP_NAME}-${SUFFIX}
+      template:
+        metadata:
+          name: ${COMP_NAME}-${SUFFIX}
+          labels:
+            name: ${COMP_NAME}-${SUFFIX}
+        spec:
+          volumes:
+            - name: mssql-data
+              persistentVolumeClaim:
+                claimName: ${COMP_NAME}-${SUFFIX}-data
+          containers:
+            - name: ${COMP_NAME}-${SUFFIX}
+              image: ""
+              ports:
+                - containerPort: 1433
+                  protocol: TCP
+              env:
+                - name: ACCEPT_EULA
+                  value: "Y"
+                - name: MSSQL_SA_PASSWORD
+                  valueFrom:
+                    secretKeyRef:
+                      name: ${COMP_NAME}-${SUFFIX}-secret
+                      key: MSSQL_SA_PASSWORD
+              resources:
+                limits:
+                  memory: ${MEMORY_LIMIT}
+              volumeMounts:
+                - mountPath: /var/opt/mssql
+                  name: mssql-data
+  - kind: PersistentVolumeClaim
+    apiVersion: v1
+    metadata:
+      name: ${COMP_NAME}-${SUFFIX}-data
+    spec:
+      accessModes:
+        - ReadWriteOnce
+      resources:
+        requests:
+          storage: ${VOLUME_CAPACITY}
+parameters:
+  - name: APP_NAME
```

Comment:
You're using COMP_NAME instead of this correct?

Reply:
Well, we use APP_NAME as the group (ie pims) and then we append the COMP_NAME (ie database) to form the full name -> pims-database

We can always tweak how we generate the names